packages/core/src/indexing-store/historical.ts

```diff
@@ -334,36 +366,38 @@ export const getHistoricalStore = ({
       tableName: string;
       id: UserId;
     }) => {
-      const id = structuredClone(_id);
-      const cacheKey = getCacheKey(id, tableName);
+      if (shouldFlush()) await flush({ isFullFlush: false });
```

Comment:
Is there a reason you moved this to the top of the function in this last commit?

Reply:
I think it should be outside of the `wrap`